pkg/reloader/reloader.go

```diff
@@ -1,5 +1,56 @@
 // Package reloader contains helpers to trigger reloads of Prometheus instances
 // on configuration changes and to substitute environment variables in config files.
+//
+// While you want to watch change event of the specified config files
```

Comment:
>  the way to use the feature in Thanos is simply by importing this package

This is quite straightfoward so can we rephrase this paragraph to something like:

```
Reloader type is useful when you want to:
* Watch on changes against certain file e.g (`cfgFile`) .
* Optionally, specify different different output file for watched `cfgFile` (`cfgOutputFile`). This will also try decompress the `cfgFile` if needed and substitute ALL the envvars using Kubernetes substitution format: (`$(var)`)
* Watch on changes against certain directories (`ruleDires`). 

Once any of those two changes Prometheus on given `reloadURL` will be notified, causing Prometheus to reload configuration and rules.
```